src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Usage/DetectPLINQNops.Fixer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Microsoft.NetCore.Analyzers.Usage
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = DetectPLINQNopsAnalyzer.RuleId), Shared]
+    public sealed class DetectPLINQNopsFixer : CodeFixProvider
+    {
+        private static readonly string[] removableEnds = new string[] { "ToList", "ToArray", "AsParallel" };
+
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(DetectPLINQNopsAnalyzer.RuleId);
+
+        public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+            if (root.FindNode(context.Span) is not InvocationExpressionSyntax declaration)
+            {
+                return;
+            }
+
+            context.RegisterCodeFix(
+                new AsPArallelCodeAction(
+                    title: MicrosoftNetCoreAnalyzersResources.RemoveRedundantCall,
+                    createChangedSolution: c => RemoveAsParallelCall(context.Document, declaration, c),
+                    equivalenceKey: MicrosoftNetCoreAnalyzersResources.RemoveRedundantCall),
+                context.Diagnostics);
+        }
+
+        private static async Task<Solution> RemoveAsParallelCall(Document document, InvocationExpressionSyntax invocationExpression, CancellationToken cancellationToken)
+        {
+            var originalSolution = document.Project.Solution;
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            ExpressionSyntax possibleInvocation = invocationExpression;
+
+            do
+            {
+                var newExpression = ((possibleInvocation as InvocationExpressionSyntax)!.Expression as MemberAccessExpressionSyntax)!.Expression;
```

Comment:
Why are you using the `as`-cast to discard the null-case (call to `!`) right after?

Either you want:

```suggestion
                var newExpression = ((possibleInvocation as InvocationExpressionSyntax)?.Expression as MemberAccessExpressionSyntax)?.Expression;
```

or

```suggestion
                var newExpression = ((MemberAccessExpressionSyntax)((InvocationExpressionSyntax)possibleInvocation).Expression)!.Expression;
```